### PR TITLE
Change clog to use a published nuget package

### PR DESCRIPTION
**Pull request by ThadHouse**
Much cleaner and easier to update. In the future would likely grab the headers and configuration needed as well remotely, but for now this is good

### CODE REVIEW

**Comment by ThadHouse** [comment]
This is good to go now. Docker build is also running as well, so that won't break.

File: Dockerfile [hunk 1]
```diff
@@ -11,6 +11,7 @@ FROM    source as build
 WORKDIR /src/Debug
 RUN     chmod +x /src/scripts/install-powershell-docker.sh
 RUN     /src/scripts/install-powershell-docker.sh
+ENV	    PATH="/root/.dotnet/tools:${PATH}"
```
Comment (nibanks) [inline Dockerfile:14]: Can you line this up with the rest please?

Comment (ThadHouse) [reply Dockerfile]: 😢 VS Code had made the first character a tab rather then a space... lol

File: src/core/CMakeLists.txt [hunk 2]
```diff
@@ -1,8 +1,15 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 
+message(STATUS ${PROJECT_SOURCE_DIR})
+file(GLOB_RECURSE TESTING_CMAKE_DIR LIST_DIRECTORIES true ${PROJECT_SOURCE_DIR}/*)
+
+message(STATUS "FILES: ${TESTING_CMAKE_DIR}")
+
+message(STATUS "${CMAKE_CURRENT_SOURCE_DIR}")
```
Comment (nibanks) [inline src/core/CMakeLists.txt:9]: Necessary?

Comment (ThadHouse) [reply src/core/CMakeLists.txt]: Nope

File: .azure/templates/build-config-user.yml [hunk 3]
```diff
@@ -22,6 +22,13 @@ jobs:
     ${{ if eq(parameters.platform, 'windows') }}:
       path: msquic
 
+  - task: UseDotNet@2
```
Comment (nibanks) [inline .azure/templates/build-config-user.yml:25]: This adds nearly 2 minutes to our build time. Is it really necessary?

Comment (ThadHouse) [reply .azure/templates/build-config-user.yml:25]: I can disable it on windows. It is required for linux, where it only takes 6 seconds...

Comment (nibanks) [reply .azure/templates/build-config-user.yml:25]: Please do so.

Comment (ThadHouse) [reply .azure/templates/build-config-user.yml:25]: Done

File: .azure/templates/build-config-user.yml [hunk 4]
```diff
@@ -22,6 +22,15 @@ jobs:
...
+
```
Comment (nibanks) [inline .azure/templates/build-config-user.yml:26]: nit: extra line?

File: scripts/build.ps1 [hunk 5]
```diff
@@ -292,6 +292,8 @@ function CMake-Build {
 #                     Main Execution                         #
 ##############################################################
 
+clog --installDirectory (Join-Path $RootDir build clog)
```
Comment (nibanks) [inline scripts/build.ps1:295]: We want to support cmake-only (no pwsh) builders. This won't help them. Should we make this part of the prepare-machine steps (and update the cmake build instructions)?

Comment (nibanks) [reply scripts/build.ps1:295]: Or make cmake do this?

Comment (ThadHouse) [reply scripts/build.ps1]: Done. Moved into cmake.

**Pull request merged**